### PR TITLE
If no tag in the repo, then show the commit hash

### PR DESCRIPTION
**Pull request by dneto0**
Also, use subprocess.Popen and p.communicate to get
better portability for Windows.

### CODE REVIEW

File: utils/update_build_version.py [hunk 1]
```diff
@@ -30,13 +30,32 @@
 
 OUTFILE = 'build-version.inc'
 
+
+def command_output(cmd, dir):
+    """Runs a command in a directory and returns its standard output stream.
+
+    Raises a RuntimeError if the command fails to launch or otherwise fails.
+    """
+    p = subprocess.Popen(cmd, stdout=subprocess.PIPE, cwd=dir)
```
Comment (AWoloszyn) [inline utils/update_build_version.py]: To avoid spamming stderr if git-describe fails, you could stderr=subprocess.PIPE as well.

**Comment by dneto0** [comment]
All feedback addressed.

**Comment by AWoloszyn** [comment]
+2

File: utils/update_build_version.py [hunk 2]
```diff
@@ -30,13 +30,37 @@
...
+
+    Raises a RuntimeError if the command fails to launch or otherwise fails.
+    """
+    p = subprocess.Popen(cmd,
+                         cwd=dir,
+                         stdout=subprocess.PIPE,
+                         stderr=subprocess.PIPE)
+    (stdout, _) = p.communicate()
+    if p.returncode !=0:
+        raise RuntimeError("Failed to run %s in %s" % (cmd, dir))
```
Comment (dekimir) [inline utils/update_build_version.py]: Optional: might be useful to include stderr content in the message.

**Comment by dneto0** [comment]
Rebased and pushed into master as 8f7357ce

**Pull request closed without merging**